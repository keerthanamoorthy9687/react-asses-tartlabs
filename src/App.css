.App {
  text-align: justify;
  font-size: 20px;
  font-family: "Times New Roman", Times, serif;

  min-height: 100vh;
}

.App-Header {
  display: flex;
  flex-direction: column;
  justify-content: center;

  border-bottom: 1.5px solid #fff;
}

.App-Container {
  padding: 50px 50px;
}

.grid {
  display: grid;
}

.grid-cols-1 {
  grid-template-columns: repeat(1, minmax(0, 1fr));
}
.grid-cols-2 {
  grid-template-columns: repeat(2, minmax(0, 1fr));
}

.title {
  color: #fff;
  font-weight: 500;
  font-size: 2rem;
  letter-spacing: 0.00735em;
  line-height: 2.5rem;
}

.navbar .navbar-brand {
  font-size: 1.6rem;
}

.truncate {
  white-space: nowrap;
}

.float-left {
  float: left;
}
.mr-1 {
  margin-right: 1rem;
}

.text-lg {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.font-bold {
  font-weight: bold;
}

.movie-detail {
  display: flex;
  justify-content: center;
}

.movie-form {
  text-align: left;
  font-size: 20px;
  color: #fff;
}

.movie-title {
  color: #fff;
  font-weight: bold;

  letter-spacing: 0.00735em;
  line-height: 2.5rem;
  display: flex;
  align-items: left;
  justify-content: left;
  margin-bottom: 2rem;
}

.mr-4 {
  margin-right: 1rem;
}
@media (min-width: 768px) {
  .md\:grid-cols-2 {
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }
}
