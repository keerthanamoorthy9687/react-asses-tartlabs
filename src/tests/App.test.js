import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import App from "../App";

const middlewares = [thunk];

describe("TartLabs Testing Library", () => {
  const initialState = { app: { posts: [], post: {}, loading: false } };
  const mockStore = configureStore(middlewares);
  let store;

  it('app - snapshot testing"', () => {
    store = mockStore(initialState);
    const wrapper = render(
      <Provider store={store}>
        <App />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
