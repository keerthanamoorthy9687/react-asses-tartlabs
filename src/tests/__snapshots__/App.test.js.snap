// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TartLabs Testing Library app - snapshot testing" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="App-Header py-3"
        >
          <div
            class="container"
          >
            <nav
              class="navbar navbar-expand-lg navbar-dark"
            >
              <div
                class="container-fluid"
              >
                <a
                  class="navbar-brand"
                  href="/"
                >
                  Navbar scroll
                </a>
                <button
                  aria-controls="navbarScroll"
                  aria-label="Toggle navigation"
                  class="navbar-toggler collapsed"
                  type="button"
                >
                  <span
                    class="navbar-toggler-icon"
                  />
                </button>
                <div
                  class="navbar-collapse collapse"
                  id="navbarScroll"
                >
                  <div
                    class="me-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
                    style="max-height: 100px;"
                  >
                    <a
                      class="nav-link"
                      data-rr-ui-event-key="/posts"
                      href="/posts"
                    >
                      Posts
                    </a>
                    <a
                      class="nav-link"
                      data-rr-ui-event-key="/posts/new"
                      href="/posts/new"
                    >
                      Add Posts
                    </a>
                    <div
                      class="nav-item dropdown"
                    >
                      <a
                        aria-expanded="false"
                        class="dropdown-toggle nav-link"
                        href="#"
                        id="navbarScrollingDropdown"
                        role="button"
                        tabindex="0"
                      >
                        Others
                      </a>
                    </div>
                  </div>
                  <form
                    class="d-flex"
                  >
                    <input
                      aria-label="Search"
                      class="me-2 form-control"
                      placeholder="Search"
                      type="search"
                    />
                    <button
                      class="btn btn-success"
                      type="button"
                    >
                      Search
                    </button>
                  </form>
                </div>
              </div>
            </nav>
          </div>
        </div>
        <div
          class="App-Container container"
        >
          <div
            class="bg-white rounded shadow p-4 overflow-hidden mb-4"
          >
            <button
              class="px-2 btn btn-primary"
              type="button"
            >
              Users List
            </button>
             
            <button
              class="px-2 btn btn-outline-secondary"
              type="button"
            >
              Add Post
            </button>
             
            <button
              class="px-2 btn btn-outline-success"
              type="button"
            >
              Posts List
            </button>
             
            <button
              class="px-2 btn btn-outline-warning"
              type="button"
            >
              Users Posts
            </button>
             
            <button
              class="px-2 btn btn-outline-danger"
              type="button"
            >
              Comments List
            </button>
             
            <button
              class="px-2 btn btn-outline-info"
              type="button"
            >
              Tag List
            </button>
             
            <button
              class="px-2 btn btn-outline-dark"
              type="button"
            >
              Post by Tag
            </button>
             
          </div>
          <div
            class="mb-5"
          >
            <div
              class="grid gap-4 grid-cols-2 md:grid-cols-2"
            />
          </div>
          <div
            class="bg-white rounded shadow p-4 overflow-hidden mb-4 "
          >
            Created by 
            <span
              class="font-bold"
            >
              Developers
            </span>
             for
             
            <span
              class="font-bold"
            >
              Developers
            </span>
            . © 2022
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="App-Header py-3"
      >
        <div
          class="container"
        >
          <nav
            class="navbar navbar-expand-lg navbar-dark"
          >
            <div
              class="container-fluid"
            >
              <a
                class="navbar-brand"
                href="/"
              >
                Navbar scroll
              </a>
              <button
                aria-controls="navbarScroll"
                aria-label="Toggle navigation"
                class="navbar-toggler collapsed"
                type="button"
              >
                <span
                  class="navbar-toggler-icon"
                />
              </button>
              <div
                class="navbar-collapse collapse"
                id="navbarScroll"
              >
                <div
                  class="me-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
                  style="max-height: 100px;"
                >
                  <a
                    class="nav-link"
                    data-rr-ui-event-key="/posts"
                    href="/posts"
                  >
                    Posts
                  </a>
                  <a
                    class="nav-link"
                    data-rr-ui-event-key="/posts/new"
                    href="/posts/new"
                  >
                    Add Posts
                  </a>
                  <div
                    class="nav-item dropdown"
                  >
                    <a
                      aria-expanded="false"
                      class="dropdown-toggle nav-link"
                      href="#"
                      id="navbarScrollingDropdown"
                      role="button"
                      tabindex="0"
                    >
                      Others
                    </a>
                  </div>
                </div>
                <form
                  class="d-flex"
                >
                  <input
                    aria-label="Search"
                    class="me-2 form-control"
                    placeholder="Search"
                    type="search"
                  />
                  <button
                    class="btn btn-success"
                    type="button"
                  >
                    Search
                  </button>
                </form>
              </div>
            </div>
          </nav>
        </div>
      </div>
      <div
        class="App-Container container"
      >
        <div
          class="bg-white rounded shadow p-4 overflow-hidden mb-4"
        >
          <button
            class="px-2 btn btn-primary"
            type="button"
          >
            Users List
          </button>
           
          <button
            class="px-2 btn btn-outline-secondary"
            type="button"
          >
            Add Post
          </button>
           
          <button
            class="px-2 btn btn-outline-success"
            type="button"
          >
            Posts List
          </button>
           
          <button
            class="px-2 btn btn-outline-warning"
            type="button"
          >
            Users Posts
          </button>
           
          <button
            class="px-2 btn btn-outline-danger"
            type="button"
          >
            Comments List
          </button>
           
          <button
            class="px-2 btn btn-outline-info"
            type="button"
          >
            Tag List
          </button>
           
          <button
            class="px-2 btn btn-outline-dark"
            type="button"
          >
            Post by Tag
          </button>
           
        </div>
        <div
          class="mb-5"
        >
          <div
            class="grid gap-4 grid-cols-2 md:grid-cols-2"
          />
        </div>
        <div
          class="bg-white rounded shadow p-4 overflow-hidden mb-4 "
        >
          Created by 
          <span
            class="font-bold"
          >
            Developers
          </span>
           for
           
          <span
            class="font-bold"
          >
            Developers
          </span>
          . © 2022
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
